repo/format/format_manager_test.go

```diff
@@ -169,6 +171,179 @@ func TestInitialize(t *testing.T) {
 		format.ErrAlreadyInitialized)
 }
 
+func TestInitializeWithRetention(t *testing.T) {
+	ctx := testlogging.Context(t)
+
+	ta := faketime.NewClockTimeWithOffset(0)
+	nowFunc := ta.NowFunc()
+
+	st := blobtesting.NewVersionedMapStorage(nowFunc).(cache.Storage)
+	blobCache := format.NewMemoryBlobCache(nowFunc)
+	mode := blob.Governance
+	period := time.Hour * 48
+
+	// success
+	require.NoError(t, format.Initialize(
+		ctx,
+		st,
+		&format.KopiaRepositoryJSON{},
+		rc,
+		format.BlobStorageConfiguration{
+			RetentionMode:   mode,
+			RetentionPeriod: period,
+		},
+		"some-password",
+	))
+
+	mgr, err := format.NewManagerWithCache(ctx, st, cacheDuration, "some-password", nowFunc, blobCache)
+	require.NoError(t, err, "getting format manager")
+
+	// New retention parameters should be available from the format manager.
+	blobCfg := mustGetBlobStorageConfiguration(t, mgr)
+	assert.Equal(t, mode, blobCfg.RetentionMode)
+	assert.Equal(t, period, blobCfg.RetentionPeriod)
+
+	// Attempting to touch the blobs the format manager writes should return
+	// errors as they should have retention enabled. Mod time adjustment (duration
+	// param) doesn't matter in this context.
+	_, err = st.TouchBlob(ctx, format.KopiaRepositoryBlobID, time.Minute)
+	assert.ErrorIs(t, err, blobtesting.ErrBlobLocked, "Altering locked repo blob should fail")
+
+	_, err = st.TouchBlob(ctx, format.KopiaBlobCfgBlobID, time.Minute)
+	assert.ErrorIs(t, err, blobtesting.ErrBlobLocked, "Altering locked blob storage config should fail")
+
+	// Advance the clock to just before the retention period expires and check
+	// if we can modifiy the blob. Blobs should still be protected so TouchBlob
+	// should still fail.
+	ta.Advance((time.Hour * 48) - time.Minute)
+	_, err = st.TouchBlob(ctx, format.KopiaRepositoryBlobID, time.Minute)
+	assert.ErrorIs(t, err, blobtesting.ErrBlobLocked, "Altering locked repo blob should fail")
+
+	_, err = st.TouchBlob(ctx, format.KopiaBlobCfgBlobID, time.Minute)
```

Comment:
@Shrekster separate from this PR.
The semantics of `TouchBlob` do not match my mental model and understanding of object locking. Having said that, it is sufficient for testing this particular set of changes.

Reply:
Agreed. We might be able to deprecate `TouchBlob` API with some caching changes, I believe that is the only storage piece that ever needs it.

My only concern with extended reliance on `TouchBlob` here was that it builds up the technical debt, but as @ashmrtn explained it is too much work for this PR otherwise.